File: Backend/sentilId-model/backend/routes/evaluate.js
const express = require("express");
const axios = require("axios");

const router = express.Router();

// POST /evaluate
router.post("/", async (req, res) => {
try {
const { features } = req.body;

kotlin
Copy
Edit
if (!features || !Array.isArray(features) || features.length !== 7) {
  return res.status(400).json({ error: "Invalid input: Expected 7 features." });
}

// Send request to Flask model server
const flaskResponse = await axios.post("http://localhost:5000/predict", {
  features,
});

const { prediction, probability } = flaskResponse.data;

return res.json({
  label: prediction === 1 ? "human" : "bot",
  confidence: probability,
});
} catch (error) {
console.error("Error evaluating:", error.message);
return res.status(500).json({ error: "Failed to evaluate behavior." });
}
});

File: Backend/sentilId-model/backend/routes/evaluate.py
from flask import Blueprint, request, jsonify
import numpy as np
from logic.classifier import model, scaler

evaluate_bp = Blueprint("evaluate", __name__)

@evaluate_bp.route("/", methods=["POST"])
def predict():
    try:
        data = request.json.get("features")
        if not data:
            return jsonify({"error": "No input features provided"}), 400

        X = np.array(data).reshape(1, -1)
        X_scaled = scaler.transform(X)
        prob = model.predict(X_scaled)[0][0]
        prediction = int(prob >= 0.5)

        return jsonify({
            "probability": float(prob),
            "prediction": prediction
        })
    except Exception as e:
        return jsonify({"error": str(e)}), 500
